proto/redpanda/api/dataplane/v1alpha1/topic.proto

```diff
@@ -15,34 +15,39 @@ message Topic {
     ConfigSource source = 4;
     bool read_only = 5;
     bool sensitive = 6;
+    // If no config value is set at the topic level, it will inherit the value
+    // set at the broker or cluster level. Name is the corresponding config
+    // key whose value is inherited. Source indicates whether the inherited
+    // config is default, broker, etc.
     repeated ConfigSynonym config_synonyms = 7;
+    // Config documentation.
     optional string documentation = 8;
   }
 }
 
 message CreateTopicRequest {
   message Topic {
     message Config {
-      // Name is a topic level config key (e.g. segment.bytes).
+      // A topic-level config key (e.g. segment.bytes).
       string name = 1 [
         (buf.validate.field).string.min_len = 1,
         (buf.validate.field).string.max_len = 255,
         (buf.validate.field).string.pattern = "^[a-z0-9.]+$"
       ];
 
-      // Value is a topic level config value (e.g. 1073741824)
+      // A topic-level config value (e.g. 1073741824).
       optional string value = 2;
     }
 
     message ReplicaAssignment {
-      // Partition is a partition to create.
+      // A partition to create.
       int32 partition_id = 1;
 
-      // Replicas are the broker IDs the partition must exist on.
+      // The broker IDs the partition replicas are assigned to.
       repeated int32 replica_ids = 2;
     }
 
-    // Name is the topic's name.
+    // Name of topic.
```

Comment:
@weeco I also wanted to follow up on some of my [questions](https://redpandadata.slack.com/archives/C058KAULETE/p1711015914132249?thread_ts=1710971531.153559&cid=C058KAULETE) regarding the `partition_count`, `replication_factor` and `replica_assignments` annotations in the next several lines. What is there right now seems a bit contradictory to me but perhaps it's my lack of knowledge

Reply:
`-1` is correct in that case, not null. This comment is wrong. If `null` is set the cluster defaults for partition_count and replication_factor will be applied in fact.

The validate constraints must be updated like so though. I haven't done yet to avoid further merge conflicts, so would be nice if you update/add it as follows:

```proto
    // PartitionCount is how many partitions to give a topic. This must
    // be null if specifying partitions manually (see ReplicaAssignment)
    // or, to use the cluster default partitions.
    optional int32 partition_count = 2 [
      (buf.validate.field).int32.gte = -1
    ];

    // ReplicationFactor is how many replicas every partition must have.
    // This must be null if specifying partitions manually (see ReplicaAssignment)
    // or, to use the cluster default replication factor.
    optional int32 replication_factor = 3 [
      (buf.validate.field).int32.gte = -1,
      (buf.validate.field).int32.lte = 5
    ];
```

Reply:
@weeco To set to null should they do something specifically like `"partition_count": null` or do they just leave it out of the request body entirely?

Reply:
Both should work